### ISSUE TITLE
which ruby doesn't find installed rbenv versions

### ISSUE THREAD

**Issue by lisamcho** [sstephenson/rbenv#792]
Hi,
I installed RBenv via Homebrew. I believe I have configured the path correctly, and when I run "rbenv version" I get the version of Ruby I meant to install (2.1.2), but when I run "which ruby" it shows the path 'usr/bin/ruby', and running ruby -v yields 1.8.2. When I try "which gem" it says there is none.
My .bash_profile is configured as follows:
export PATH=/usr/local/sbin:/usr/local/bin:$PATH
if which rbenv > /dev/null; then eval "$(rbenv init -)"; fi
export PATH="$HOME/.rbenv/bin:$PATH"

Is that a bug or did I do something wrong?

**Comment by mislav** [comment]
Hi, I believe these lines:

```
if which rbenv > /dev/null; then eval "$(rbenv init -)"; fi
export PATH="$HOME/.rbenv/bin:$PATH"
```

need to be reversed:

```sh
export PATH="$HOME/.rbenv/bin:$PATH"
if which rbenv > /dev/null; then eval "$(rbenv init -)"; fi
```

Let me know if that works!

**Comment by mislav** [comment]
After making changes to `.bash_profile`, you should open a new terminal window and you can see your current PATH with:

```
echo $PATH | tr : $'\n'
```

The line `$HOME/.rbenv/shims` should be there if everything was set up alright. Then you can do:

```
rbenv rehash
which gem
```

And the result should be "$HOME/.rbenv/shims/gem". From then, everything should work fine!

**Comment by lisamcho** [comment]
Hmmm, doesn't seem to work. 'ruby -v' still points to 1.8.2.
I run 
which gem

and get: no gem in /Users/lmcho/.rbenv/shims /Users/lmcho/.rbenv/bin /usr/local/sbin /usr/local/bin /bin /sbin /usr/bin /usr/sbin.

So theoretically it is looking in the right folder. I doublechecked the .rbenv folder also. In .rbenv/versions, there is 2.1.2 and 2.2.2. But in the .rbenv/shims folder it is empty.

Does anything jump out at you as being wrong? The problem could also be that I am trying to revive an ancient laptop as my second dev machine (using Tigerbrew - a fork of Homebrew that runs on OSX10.4). Thanks!

**Comment by mislav** [comment]
If you run `rbenv root`, does it say "/Users/lmcho/.rbenv"? Have you tried `rbenv rehash`? Is the shims folder still empty after that?

What is the output of

```
ls `rbenv root`/versions/*/bin
```

**Comment by lisamcho** [comment]
I run rbenv root.
Result: /Users/lmcho/.rbenv

I run rbenv rehash.
Shims folder still empty.

Output of the last command is as follows:
ls: /Users/lmcho/.rbenv/versions/*/bin: No such file or directory

**Comment by mislav** [comment]
The last command indicates that the versions of Ruby you've seen in `versions` aren't properly installed.  If they were, they should each have their `bin` directories.

How did you install Ruby 2.1.2 and 2.2.2 in the first place? Can you try installing them again with `rbenv install 2.2.2`? (Install ruby-build if you haven't before: `brew install ruby-build` to install rubies.)

**Comment by lisamcho** [comment]
OK, maybe this is the problem. I didn't notice there were warnings during the install...although it does say installed at the end. I am using a weird fork of homebrew (but it seems to be a wellsupported one).

/usr/local/bin/ruby-build: line 16: set: -E: invalid option
set: usage: set [--abefhkmnptuvxBCHP] [-o option] [arg ...]
/usr/local/bin/ruby-build: line 16: set: -E: invalid option
set: usage: set [--abefhkmnptuvxBCHP] [-o option] [arg ...]
usage: basename string [suffix]
       basename [-a] [-s suffix] string [...]
Downloading ...
/usr/local/bin/ruby-build: line 199: pushd: (ruby-2.2.2): No such file or directory
Installing (ruby-2.2.2)...
/usr/local/bin/ruby-build: line 188: popd: directory stack empty
Installed ruby-2.2.2 to /Users/lmcho/.rbenv/versions/2.2.2

**Comment by mislav** [comment]
Ouch, it doesn't seem that ruby-build is compatible with your version of OS X. It seems like a bash problem. What is your `bash --version`?

**Comment by lisamcho** [comment]
Ancient. Don't laugh.... The machine is a 2004 iBook G4, running...

GNU bash, version 2.05b.0(1)-release (powerpc-apple-darwin8.0)
Copyright (C) 2002 Free Software Foundation, Inc.

**Comment by mislav** [comment]
That old version of bash might be a problem. I've never coded scripts for such an old version of bash. Can tigerbrew maybe help you install a newer version of bash? Preferrably bash 3 or newer. Once the new bash is in `/usr/local/bin/bash`, you can try executing `ruby-build` again and see if that helped.

**Comment by lisamcho** [comment]
Hmmm. OK, I am now running a recent version of bash (GNU bash, version 4.3.42(1)-release (powerpc-apple-darwin8.11.0).

I uninstalled the ruby versions that had been installed properly, then I uninstalled rbenv and ruby-build and reinstalled them just to make sure.  I tried to install two different versions of ruby. I get "BUILD FAILED Inspect or clean up the working tree" for both. Any idea what is going on now? Logs say...

2.2.3:
/usr/libexec/gcc/powerpc-apple-darwin8/4.0.1/ld: Undefined symbols:
_rb_enc_set_base
_rb_enc_set_dummy
_rb_encdb_alias
_rb_encdb_declare
_rb_encdb_dummy
_rb_encdb_replicate
_rb_encdb_set_unicode
collect2: ld returned 1 exit status
make[1]: *** [.ext/powerpc-darwin8.0/enc/encdb.bundle] Error 1
make: *** [enc] Error 2

2.1.0:
compiling ./missing/alloca.c
./missing/alloca.c: In function 'alloca':
./missing/alloca.c:149: warning: argument 'size' doesn't match built-in prototype
./missing/alloca.c: At top level:
./missing/alloca.c:29: warning: 'SCCSid' defined but not used
compiling dmyext.c
linking miniruby
/usr/libexec/gcc/powerpc-apple-darwin8/4.0.1/ld: Undefined symbols:
_pthread_setname_np
collect2: ld returned 1 exit status
make: *** [miniruby] Error 1

**Comment by mislav** [comment]
Well, I'm glad that bash isn't giving you trouble anymore, but I was afraid this might happen. The thing is, newer releases of Ruby like the ones you're trying likely aren't tested on Tiger. They might not be able to build, as suggested by these failures.

Can you check your version of gcc? Are you able to install gcc-42 from Tigerbrew (if a formula like that is available) and use CC=/usr/local/bin/gcc-4.2 for installation?

Also you might want to search the web if anyone has experience building Ruby on Tiger. I wish I could help you, but there's little I can suggest without having a Tiger machine at hand. :(

**Comment by mislav** [comment]
@lisamcho Since this is a problem of ruby-build on Tiger, do you mind posting future updates to a new issue in https://github.com/sstephenson/ruby-build/issues? rbenv itself really doesn't have anything to do with compiling Rubies, it just switches versions.

Please update us if you manage to build on Tiger. Thanks!
